### PR TITLE
Add IOWA case study

### PR DESCRIPTION
**Pull request by ebidel**
R: @petele @PaulKinlan @jeffposnick

### CODE REVIEW

**Comment by PaulKinlan** [comment]
I don't know our new infra and I can't test on staging anymore, so I'll leave it to Pete.  Generally it looks good.

File: src/content/en/showcase/2016/iowa2016.md [hunk 1]
```diff
@@ -0,0 +1,473 @@
+project_path: /web/_project.yaml
+book_path: /web/showcase/_book.yaml
+description: Building the Google I/O 2016 PWA using web components
+
+{# wf_published_on: 2016-05-17 #}
+{# wf_updated_on: 2016-05-17 #}
+{# wf_author: ericbidelman #}
+{# wf_featured_image: /web/showcase/2016/images/DIR/featured.png #}
```
Comment (petele) [inline src/content/en/showcase/2016/iowa2016.md]: You can remove this line

File: src/content/en/showcase/_book.yaml [hunk 2]
```diff
@@ -23,6 +23,8 @@ upper_tabs:
         contents:
         - title: "2016"
           section:
+          - title: Building the Google I/O 2016 Progressive Web App
+            path: /web/showcase/20156/iowa2016
```
Comment (petele) [inline src/content/en/showcase/_book.yaml]: `/web/showcase/2016/iowa2016` remove the extra 5

File: src/content/en/showcase/2016/iowa2016.md [hunk 3]
```diff
@@ -0,0 +1,473 @@
...
+IOWA is a [single page application](https://en.wikipedia.org/wiki/Single-page_application) (SPA) that's built using [web components](http://webcomponents.org/). It pre-caches content using a [service worker](http://www.html5rocks.com/en/tutorials/service-worker/introduction/), dynamically loads new pages, gracefully transitions between views, and reuses content after first load.
+
+In this case study, I'll go through some of the more interesting architectural decisions that we made for this year's frontend. If you're interested in the full source code, [check it out](github.com/GoogleChrome/ioweb2016) on Github.
+
+### Every page as a component
+
+One of the core aspects about our frontend is that it's centered around web components. In fact, every page in our SPA is a web component:
+
+    <io-home-page date="2016-05-18T17:00:00Z" app="[[app]]"></io-home-page>
+    <io-schedule-page date="2016-05-18T17:00:00Z" app="{{app}}"></io-schedule-page>
```
Comment (petele) [inline src/content/en/showcase/2016/iowa2016.md]: Django thinks the `{{` in docs is actually meant as a variable, not code. To escape use `{% templatetag openvariable %}{app}}`

File: src/content/en/showcase/2016/iowa2016.md [hunk 4]
```diff
@@ -0,0 +1,473 @@
...
+
+    <google-signin client-id="..." scopes="profile email"
+                             user="{{app.currentUser}}"></google-signin>
+
+    <iron-media-query query="(min-width:320px) and (max-width:768px)"
+                                  query-matches="{{app.isPhoneSize}}"></iron-media-query>
+
+The `<google-signin>` element updates its `user` property when users login to our app. Since that property is bound to `app.currentUser`, any page that wants to access the current user simply needs to bind to `app` and read the `currentUser` sub-property. By itself, this technique is useful for sharing state across the app. However, another benefit was that we ended up creating a **single sign-in element and reused its results across the site**. Same for the media queries. It would have been wasteful for every page to duplicate sign-in or create its own set of media queries. Instead, components which are responsible for app-wide functionality/data exist at the app-level.
+
+Note: I'm purposefully showing two-way bindings (`{{}}`) and one-way bindings
```
Comment (petele) [inline src/content/en/showcase/2016/iowa2016.md]: Same issue here for `{{`

File: src/content/en/showcase/2016/iowa2016.md [hunk 5]
```diff
@@ -0,0 +1,473 @@
...
+(`[[]]`) in this example. Use (`{{}}`) bindings when changes need to flow into
```
Comment (petele) [inline src/content/en/showcase/2016/iowa2016.md]: And here `{{`

File: src/content/en/showcase/2016/iowa2016.md [hunk 6]
```diff
@@ -0,0 +1,473 @@
...
+an element as well as propagate back out (when changes are made internally).
+Use (`[[]]`) bindings when an element just needs to read data.
+
+## Page transitions
+
+As you navigate the Google I/O web app, you'll notice it's a SPA with slick
+page transitions (_à la_ [material design](https://www.google.com/design/spec/material-design/introduction.html)).
+
+[VIDEO]
```
Comment (petele) [inline src/content/en/showcase/2016/iowa2016.md]: Video?

File: src/content/en/showcase/2016/iowa2016.md [hunk 7]
```diff
@@ -0,0 +1,473 @@
...
+
+      Polymer.AppLayout.scroll({top: 0, behavior: 'smooth',
+                                                 target: document.documentElement});
+
+      e.target.blur();  // Kick focus back to the page so user starts from the top of the doc.
+    }
+
+Another place we used the `<app-layout>` elements was for the sticky nav. As you can see in the video, it goes away when users scroll down the page and returns when scrolling back up.
+
+[VIDEO]
```
Comment (petele) [inline src/content/en/showcase/2016/iowa2016.md]: Video?

File: src/content/en/showcase/2016/iowa2016.md [hunk 8]
```diff
@@ -0,0 +1,473 @@
...
+      // page transition animations are complete.
+    },
+
+    onSubpageTransitionDone() {
+      // sub nav/tab page transitions are complete.
+    }
+
+These were useful additions for delaying work and minimizing the jank between page transitions. More on this later.
+
+<h3 id="dry">DRYing up common functionality across pages</h3>
```
Comment (petele) [inline src/content/en/showcase/2016/iowa2016.md]: You can use `### DRYing up common functionality across pages {: #dry }` if you want to use MD for titles, not a requirement but more consistent

File: src/content/en/showcase/2016/iowa2016.md [hunk 9]
```diff
@@ -0,0 +1,473 @@
...
+    </a>
+
+The smooth scroll is implemented using Polymer's [app-layout elements](https://elements.polymer-project.org/elements/app-layout). They provide out-of-the-box scrolling effects like sticky/returning top navs, drop shadows, color and background transitions, parallax effects, and smooth scrolling.
+
+    // Smooth scrolling the back to top FAB.
+    function backToTop(e) {
+      e.preventDefault();
+
+      Polymer.AppLayout.scroll({top: 0, behavior: 'smooth',
+                                                 target: document.documentElement});
```
Comment (petele) [inline src/content/en/showcase/2016/iowa2016.md]: Spacing on these lines seems a bit off?

**Comment by petele** [comment]
Staged internally for review. Couple of minor issues to be addressed but then should be good to go.

**Comment by jpmedley** [comment]
I want to look at this before it's merged. Has the content settled down?

File: src/content/en/showcase/_book.yaml (hunk 2 above)
Comment (ebidel) [inline src/content/en/showcase/_book.yaml]: Done

File: src/content/en/showcase/2016/iowa2016.md (hunk 1 above)
Comment (ebidel) [inline src/content/en/showcase/2016/iowa2016.md]: Done

File: src/content/en/showcase/2016/iowa2016.md (hunk 3 above)
Comment (ebidel) [inline src/content/en/showcase/2016/iowa2016.md]: Done

File: src/content/en/showcase/2016/iowa2016.md (hunk 4 above)
Comment (ebidel) [inline src/content/en/showcase/2016/iowa2016.md]: Done

File: src/content/en/showcase/2016/iowa2016.md (hunk 5 above)
Comment (ebidel) [inline src/content/en/showcase/2016/iowa2016.md]: Done

File: src/content/en/showcase/2016/iowa2016.md (hunk 8 above)
Comment (ebidel) [inline src/content/en/showcase/2016/iowa2016.md]: Done

File: src/content/en/showcase/2016/iowa2016.md (hunk 9 above)
Comment (ebidel) [inline src/content/en/showcase/2016/iowa2016.md]: done

**Comment by ebidel** [comment]
@jpmedley the content is final

**Pull request merged**